From d534e6ca7931f785adc9f756be68ddfe9f19daf8 Mon Sep 17 00:00:00 2001
From: Xinhuang Li <buluess.li@rock-chips.com>
Date: Thu, 7 Dec 2017 16:22:01 +0800
Subject: [PATCH] driver: video: rockchip: fix vepu build fail and can not work

1.vepu need core_clk define
2.there is no ION_HEAP function,replace it with the same definition

Change-Id: I7cf02c9e446976f0424d9489e081c9614b7e7e0c
Signed-off-by: Xinhuang Li <buluess.li@rock-chips.com>
---
 drivers/video/rockchip/vpu/mpp_dev_vepu.c  | 10 ++++++++++
 drivers/video/rockchip/vpu/mpp_dev_vepu.h  |  1 +
 drivers/video/rockchip/vpu/vpu_iommu_ion.c |  4 ++--
 3 files changed, 13 insertions(+), 2 deletions(-)

diff --git a/drivers/video/rockchip/vpu/mpp_dev_vepu.c b/drivers/video/rockchip/vpu/mpp_dev_vepu.c
index 27148b7a1896..1259055e92af 100644
--- a/drivers/video/rockchip/vpu/mpp_dev_vepu.c
+++ b/drivers/video/rockchip/vpu/mpp_dev_vepu.c
@@ -287,6 +287,8 @@ static void rockchip_mpp_vepu_power_on(struct rockchip_mpp_dev *mpp)
 		clk_prepare_enable(enc->aclk);
 	if (enc->hclk)
 		clk_prepare_enable(enc->hclk);
+	if (enc->cclk)
+		clk_prepare_enable(enc->cclk);
 }
 
 static void rockchip_mpp_vepu_power_off(struct rockchip_mpp_dev *mpp)
@@ -297,6 +299,8 @@ static void rockchip_mpp_vepu_power_off(struct rockchip_mpp_dev *mpp)
 		clk_disable_unprepare(enc->hclk);
 	if (enc->aclk)
 		clk_disable_unprepare(enc->aclk);
+	if (enc->cclk)
+		clk_disable_unprepare(enc->cclk);
 }
 
 static int rockchip_mpp_vepu_probe(struct rockchip_mpp_dev *mpp)
@@ -318,6 +322,12 @@ static int rockchip_mpp_vepu_probe(struct rockchip_mpp_dev *mpp)
 		goto fail;
 	}
 
+	enc->cclk = devm_clk_get(mpp->dev, "clk_core");
+	if (IS_ERR_OR_NULL(enc->cclk)) {
+		dev_err(mpp->dev, "failed on clk_get cclk\n");
+		goto fail;
+	}
+
 	if (of_property_read_bool(np, "mode_ctrl")) {
 		of_property_read_u32(np, "mode_bit", &enc->mode_bit);
 		of_property_read_u32(np, "mode_ctrl", &enc->mode_ctrl);
diff --git a/drivers/video/rockchip/vpu/mpp_dev_vepu.h b/drivers/video/rockchip/vpu/mpp_dev_vepu.h
index 3e0d5875d5dd..8bfbe5ab2d8b 100644
--- a/drivers/video/rockchip/vpu/mpp_dev_vepu.h
+++ b/drivers/video/rockchip/vpu/mpp_dev_vepu.h
@@ -27,6 +27,7 @@ struct rockchip_vepu_dev {
 
 	struct clk *aclk;
 	struct clk *hclk;
+	struct clk *cclk;
 
 	struct reset_control *rst_a;
 	struct reset_control *rst_h;
diff --git a/drivers/video/rockchip/vpu/vpu_iommu_ion.c b/drivers/video/rockchip/vpu/vpu_iommu_ion.c
index 0dcd238be8b1..06d7c2c27016 100644
--- a/drivers/video/rockchip/vpu/vpu_iommu_ion.c
+++ b/drivers/video/rockchip/vpu/vpu_iommu_ion.c
@@ -240,9 +240,9 @@ static int vpu_ion_alloc(struct vpu_iommu_session_info *session_info,
 	unsigned int heap_id_mask;
 
 	if (iommu_info->mmu_dev)
-		heap_id_mask = ION_HEAP(ION_VMALLOC_HEAP_ID);
+		heap_id_mask = ION_HEAP_TYPE_SYSTEM;
 	else
-		heap_id_mask = ION_HEAP(ION_CMA_HEAP_ID);
+		heap_id_mask = ION_HEAP_TYPE_DMA;
 
 	ion_buffer = kzalloc(sizeof(*ion_buffer), GFP_KERNEL);
 	if (!ion_buffer)
-- 
2.35.3

